lib/reverse_sentence.rb

```diff
@@ -1,6 +1,37 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2) where n is the length of my_sentence
+# Space complexity: O(n) where n is the length of my sentence
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return my_sentence if my_sentence == "" || my_sentence == nil
+  j = my_sentence.length - 1
+  new_string = ""
+  word = ""
+  while j >= 0
+    while j >= 0 && my_sentence[j] == " "
+      j -= 1
+      new_string += " "
+    end
+
+    while j >= 0 && my_sentence[j] != " "
+      word += my_sentence[j]
+      j -= 1
+    end
+    i = 0
+    k = word.length - 1
+    while i < k
+      temp = word[i]
+      word[i] = word[k]
+      word[k] = temp
+      i += 1
+      k -= 1
```

Comment:
I'd recommend renaming some of these single letter variables to more descriptive names.